tools/accuracy_checker/openvino/tools/accuracy_checker/evaluators/custom_evaluators/opennmt_encoder_decoder_generator_evaluator.py

```diff
@@ -0,0 +1,686 @@
+        self.store_encoder_predictions = network_info['encoder'].get('store_predictions', False)
+        self.store_decoder_predictions = network_info['decoder'].get('store_predictions', False)
+        self._encoder_predictions = [] if self.store_encoder_predictions else None
+        self._decoder_predictions = [] if self.store_decoder_predictions else None
+        self._part_by_name = {'encoder': self.encoder, 'decoder': self.decoder, 'generator': self.generator}
+        self._raw_outs = OrderedDict()
+        self.adapter = create_adapter(launcher.config.get('adapter', 'nmt'))
+
+    def predict(self, identifiers, input_data, encoder_callback=None):
+        predictions, raw_outputs = [], []
+
+        for data in input_data:
+            decode_strategy = BeamSearch(self.decoder.network_info)
+
+            src_len = np.array([len(data)])
+            h, c, memory = self.encoder.predict(identifiers, {'src': np.array([[[t]] for t in data]),
+                                                              'src_len': src_len})
+            if encoder_callback:
+                encoder_callback((h, c, memory, src_len))
+            if self.store_encoder_predictions:
+                self._encoder_predictions.append((h, c, memory, src_len))
+
+            self.decoder.init_state(h, c, memory, src_len)
+            self.decoder.tile_state(decode_strategy.beam_size)
+
+            for step in range(decode_strategy.max_length):
+                decoder_input = decode_strategy.current_predictions.view().reshape([1, decode_strategy.beam_size, 1])
+
+                decoder_output, _ = self.decoder.predict(identifiers, {'input': decoder_input})
+                log_probs = self.generator.predict(identifiers, {'input': decoder_output.squeeze()})[0]
+
+                decode_strategy.advance(log_probs)
+                any_finished = decode_strategy.is_finished.any()
+                if any_finished:
+                    decode_strategy.update_finished()
+                    if decode_strategy.done:
+                        break
+
+                select_indices = decode_strategy.select_indices.squeeze()
+                if any_finished:
+                    self.decoder.reorder_state(select_indices, ('memory', 'mem_len'))
+                if decode_strategy.beam_size > 1 or any_finished:
+                    self.decoder.reorder_state(select_indices, ('h', 'c', 'input_feed'))
+
+            prediction = np.array(decode_strategy.predictions)[..., :-1]
+            prediction = self.adapter.process({'pred': np.transpose(prediction, [2, 1, 0])}, identifiers, None)
+            predictions.append(prediction)
+
+        return raw_outputs, predictions
+
+    def reset(self):
+        self.processing_frames_buffer = []
+        if self._encoder_predictions is not None:
+            self._encoder_predictions = []
+
+    def release(self):
+        self.encoder.release()
+        self.decoder.release()
```

Comment:
do you need also release decoder1?

Reply:
Decoder1 is debugging artefact, removed.